src/deserialize/parse.rs

```diff
@@ -907,6 +926,96 @@ mod test {
         )
     }
 
+    #[test]
```

Comment:
I would say it is missing a test where the discard element is a sequence inside a sequence, like:
```
#_"random comment" [:a :b :c #_(:hello :world) :d]` then the result would be a vector with `[:a :b :c :d]`